test/RoaveTest/SecurityAdvisories/VersionConstraintSortTest.php

```diff
@@ -69,17 +68,33 @@ public function comparedConstraints(): array
             ['>=1-alpha.9,<=2-alpha.9,>3-alpha.9', '>=1-alpha.9,<=2-alpha.9', 0],
         ];
 
-        return array_combine(
-            array_map(static function (array $entry): string {
-                return '"' . $entry[0] . '" <=> "' . $entry[1] . '"';
-            }, $constraints),
-            array_map(static function (array $entry): array {
-                return [
-                    VersionConstraint::fromString($entry[0]),
-                    VersionConstraint::fromString($entry[1]),
-                    $entry[2],
-                ];
-            }, $constraints)
+        return Dict\associate(
+            Vec\map(
```

Comment:
Probably better to keep `array_map()` here, since inference seems weak

Reply:
you mean inference for the closure parameter?

Reply:
Correct

Reply:
yea, this is a psalm "bug" where it's unable to infer the type itself, and it has a special case for builtin functions, not sure why really https://psalm.dev/r/fe23cc72fa

Reply:
This is because the closure when used alongside a builtin functions, inherits the context of that function ( generics ), but when used with user-land functions, it is analyzed in it's own scope.

Reply:
ref: https://github.com/vimeo/psalm/issues/5820

Reply:
let's see if we can fix the issue in psalm first, it annoys me all the time too :p